OpenDreamRuntime/Procs/Native/DreamProcNativeSavefile.cs

```diff
@@ -9,29 +13,124 @@ internal static class DreamProcNativeSavefile {
+                break;
+            case DreamObjectSavefile.SFDreamType typeValue:
+                result += $"{new string('\t', indent)}{key} = {typeValue.TypePath}\n";
+                break;
+            case DreamObjectSavefile.SFDreamListValue listValue:
+                result += $"{new string('\t', indent)}{key} = {ExportTextInternalListFormat(listValue)}\n";
+                break;
+            case DreamObjectSavefile.SFDreamDir jsonValue:
+                if(key==".")
+                    result += "\n";
+                else
+                    result += $"{new string('\t', indent)}{key}\n";
+                break;
+            default:
+                throw new NotImplementedException($"Unhandled type {key} = {value} in ExportText()");
         }
 
-        if (!bundle.ResourceManager.TryLoadIcon(exportValue, out var icon)) {
-            throw new NotImplementedException("General support for ExportText() is not implemented");
+        if(string.IsNullOrEmpty(key) || key==".")
+            indent = -1; //don't indent the subdirs of directly accessed keys or root dir
+
+        foreach (string subkey in savefile.CurrentDir.Keys) {
+            savefile.CurrentPath = subkey;
+            result += ExportTextInternal(savefile, indent + 1);
+            savefile.CurrentPath = "../";
+        }
+        return result;
+    }
+
+    private static string ExportTextInternalListFormat(DreamObjectSavefile.SFDreamJsonValue listEntry){
+        switch(listEntry) {
+            case DreamObjectSavefile.SFDreamPrimitive primitiveValue:
+                if(primitiveValue.Value.IsNull)
+                    return "null";
+                else switch(primitiveValue.Value.Type){
+                    case DreamValue.DreamValueType.String:
+                        return $"\"{primitiveValue.Value.MustGetValueAsString()}\"";
+                    case DreamValue.DreamValueType.Float:
+                        return $"{primitiveValue.Value.MustGetValueAsFloat()}";
+                }
+                throw new NotImplementedException($"Unhandled list entry type {listEntry} in ExportTextInternalListFormat()");
+            case DreamObjectSavefile.SFDreamObjectPathValue objectValue:
+                return $"object(\"{objectValue.Path}\")";
+            case DreamObjectSavefile.SFDreamType typeValue:
+                return $"{typeValue.TypePath}";
+            case DreamObjectSavefile.SFDreamListValue listValue:
+                string result = "list(";
+
+                    for(int i=0; i<listValue.AssocKeys.Count; i++){
+                        if(listValue.AssocData != null && listValue.AssocData[i] != null)
+                            result += ExportTextInternalListFormat(listValue.AssocKeys[i])+" = "+ExportTextInternalListFormat(listValue.AssocData[i]!);
+                        else
+                            result += ExportTextInternalListFormat(listValue.AssocKeys[i]);
+                        result += ",";
+                    }
```

Comment:
This is over-indented
```suggestion
                   for(int i=0; i<listValue.AssocKeys.Count; i++){
                    if(listValue.AssocData != null && listValue.AssocData[i] != null)
                        result += ExportTextInternalListFormat(listValue.AssocKeys[i])+" = "+ExportTextInternalListFormat(listValue.AssocData[i]!);
                    else
                        result += ExportTextInternalListFormat(listValue.AssocKeys[i]);
                    result += ",";
                }
```